src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_CAPACITY = 10;
+    private static final float GROW_VALUE = 1.5f;
+    private T[] values;
+    private int size;
+
+    public ArrayList() {
+        values = (T[])new Object[START_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        values[size] = value;
+        size++;
+        grow();
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index must be less then size!");
+        } else if (index == size) {
+            add(value);
+        } else {
+            T oldValue = values[index];
+            values[index] = value;
+            add(oldValue, index + 1);
```

Comment:
You should move all items one position to the right, instead of adding oldValue to the end of the array. Consider using System.arraycopy here. Also remember to check storage size

Reply:
> You should move all items one position to the right, instead of adding oldValue to the end of the array. Consider using System.arraycopy here. Also remember to check storage size

oldvalue will not be added at the and of array. it will be added on poistion index+1. 
recusively all elements will move to the right, and the size check will be performed when adding a value to a new cell.